addons/knobs/src/components/Panel.js

```diff
@@ -106,6 +108,10 @@ export default class Panel extends React.Component {
     this.props.channel.emit('addon:knobs:reset');
   }
 
+  update() {
+    this.props.channel.emit('addon:knobs:update2');
```

Comment:
Not a fan of this kind of naming =(

Reply:
Absolutely, this should be changed. This PR is definitely not complete or even good. All I am asking if the community feels there is value in an `update` function to complete the one-way data flow loop. Often my state lives in a state management library like Redux/Mobx or a form library like react-final-form/redux-form/formik, etc or parent components, so I wanted to use storybook knobs to test the complete cycle. What do you think of this idea?

Reply:
I don't like the fact that it introduces multiple sources of truth (component callbacks and UI knobs)

Maybe some of the [State addons](https://storybook.js.org/addons/addon-gallery/#state) 
or just recompose's [`withState`](https://github.com/acdlite/recompose/blob/master/docs/API.md#withstate) would solve your usecase better?

Reply:
When I am making a re-usable React component, like a simple TextField for example, it seems there are two parts to the interface that I need to consider when I am developing it. 

  1) the props that it receives (and how it updates based on changes) and 
  2) the callbacks that it fires when it is interacted with.

I need to ensure that it updates when a new value is passed in as a prop, and I need to make sure it passes the changed value in a callback when I type in the TextField.

Storybook Knobs does a great job highlighting the first part, and the actions addon is good at highlighting the second, but they remain disconnected. Thank you for the other suggestions. It seems they complete the React data cycle, but without the added visibility of a panel of knobs to also interact with the components.

I don't believe it is introducing any new state, the `update` function simply changes the pre-existing knobStore. The panel of knob controls and the actual component being showcased both receive their values from the knobStore. It is all one source of truth, just two ways to view it and change it.

Anyway, I don't think I have convinced you. Thank you for your efforts with storybook. It is a fantastic project. Maybe I will make my own addon.

Reply:
I think this brings up a good problem and I'm actually quite surprised the community hasn't built a stateful knobs addon yet.

You're probably better off forking this addon and making some architectural changes as to how the data binding is made, rather than adding a patch that kind of does what you need it to do.